packages/nest/src/generators/utils/normalize-options.ts

```diff
@@ -5,20 +5,31 @@ import type {
   NormalizedOptions,
   UnitTestRunner,
 } from './types';
+import { determineArtifactNameAndDirectoryOptions } from '@nx/devkit/src/generators/artifact-name-and-directory-utils';
 
-export function normalizeOptions(
+export async function normalizeOptions(
   tree: Tree,
+  artifactType: string,
+  callingGenerator: string,
   options: NestGeneratorOptions
-): NormalizedOptions {
-  const { sourceRoot } = readProjectConfiguration(tree, options.project);
+): Promise<NormalizedOptions> {
+  const { directory, fileName, nameAndDirectoryFormat } =
+    await determineArtifactNameAndDirectoryOptions(tree, {
+      callingGenerator,
+      artifactType,
+      name: options.name,
+      directory: options.directory,
```

Comment:
Also need to pass `derivedDirectory: options.directory`. The Nest generators are like the React ones where `directory` is relative to project's source root, so `nx g class foo --project test --directory a/b` will generate `test/src/lib/a/b/foo.ts`.